Mathlib/CategoryTheory/Abelian/Transfer.lean

```diff
@@ -198,4 +212,150 @@ def abelianOfEquivalence {C : Type u₁} [Category.{v} C] [Preadditive C] [HasFi
+A ---> R(J)                 L(A) -----> J <--------
+      /                                /          |
+     /                                /           |
+    /  g                           by adjunction  |
+   /                                /             |
+  /                                /         by injectivity
+X                              L(X)               |
+|                               |L.map f          |
+v                               v                 |
+Y                              L(Y) ---------------
+```
+-/
+
+/--
+Let `L(A) ⟶ J` be an injective presentation.
+If `g : X → R(J)` and `f : X → Y` is mono in `𝓐`, then there is an morphism `L(Y) → J`:
+* Since `L` preserves finite limits, `L(f)` is mono
+* If `L ⊣ R`, then `g` gives a `L(X) ⟶ J`
+* we then factor `X ⟶ R(J)` into `L(f)` and `L(Y) ⟶ J`
+-/
+def toInjectiveObject [PreservesFiniteLimits L] {A X Y : 𝒜} (a : InjectivePresentation <| L.obj A)
+    (g : X ⟶ adjointObjectOfInjectivePresentation L R a) (f : X ⟶ Y) [Mono f] :
+    L.obj Y ⟶ a.J :=
+  let i1 := a.injective.factors
+  (i1 ((adj.homEquiv X <| a.J).symm g) (L.map f)).choose
+
+lemma toInjectiveObject_spec [PreservesFiniteLimits L] {A X Y : 𝒜}
+    (a : InjectivePresentation <| L.obj A)
+    (g : X ⟶ adjointObjectOfInjectivePresentation L R a) (f : X ⟶ Y) [Mono f] :
+    L.map f ≫ toInjectiveObject adj a g f =
+    (adj.homEquiv X <| a.J).symm g :=
+  let i1 := a.injective.factors
+  (i1 ((adj.homEquiv X <| a.J).symm g) (L.map f)).choose_spec
+
+/--
+Let `L(A) ⟶ J` be an injective presentation.
+If `g : X → R(J)` and `f : X → Y` is mono in `𝓐`, then there is an morphism `L(Y) → J` as in
+`toInjectiveUnder`, then we obtain a map `Y ⟶ R(J)` via adjunction
+-/
+def adjointToInjective [PreservesFiniteLimits L] {A X Y : 𝒜}
+    (a : InjectivePresentation <| L.obj A)
+    (g : X ⟶ adjointObjectOfInjectivePresentation L R a) (f : X ⟶ Y) [Mono f] :
+    Y ⟶ adjointObjectOfInjectivePresentation L R a :=
+  adj.homEquiv _ _ <| toInjectiveObject adj a g f
+
+lemma adjointToInjective_spec [PreservesFiniteLimits L] {A X Y : 𝒜}
+    (a : InjectivePresentation <| L.obj A)
+    (g : X ⟶ adjointObjectOfInjectivePresentation L R a) (f : X ⟶ Y) [Mono f] :
+    f ≫ adjointToInjective adj a g f = g := by
+  have := toInjectiveObject_spec adj a g f
+  rw [← adj.homEquiv_apply_eq] at this
+  rw [← this]
+  simp only [adjointToInjective, toInjectiveObject, Adjunction.homEquiv_counit, Functor.id_obj,
+    Adjunction.homEquiv_unit, Functor.comp_obj, Functor.map_comp, Adjunction.unit_naturality_assoc,
+    Category.assoc, Adjunction.counit_naturality, Adjunction.left_triangle_components_assoc]
+  generalize_proofs h1 h2
+  congr 4
+  ext
+  rw [h1.choose_spec]
+
+lemma injective_adjointObjectOfInjectivePresentation_of_adj [PreservesFiniteLimits L] {A : 𝒜}
+    (a : InjectivePresentation <| L.obj A) :
+    Injective (adjointObjectOfInjectivePresentation L R a) where
+  factors _ _ _ := ⟨_, adjointToInjective_spec adj a _ _⟩
```

Comment:
It would be nice to state separately that if `M : ℬ` is injective, then `R.obj M` is also injective (under the assumption that `L` preserves monomorphisms). (This is related to `CategoryTheory.LiftingProperties.Adjunction`, even though it may not be necessary to use that...) Then, the core argument here seems to be that if `A : 𝒜`, and `L.obj A` embeds in an (injective) object `I : ℬ`, then the adjoint map `A ⟶ R.obj I` is also a mono (essentially what you prove in `mono_toUnder`, but there is a more direct proof relying directly on the adjunction, without the need to introduce the kernel). It seems to me that doing so would significantly shorten the length of the code.

Reply:
by avoiding `kernel`, it turns out assuming abelian-ness of categories are unnecessary! Thanks